examples/example_evo_regression_numpy.py

```diff
@@ -0,0 +1,197 @@
+"""
+
+import functools
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.constants
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+# %%
+# We first define target functions. For illustration purposes, we
+# define two functions which present different levels of difficulty
+# for the search.
+
+
+def f_target_easy(x):
+    return x[:, 0] ** 2 + 2 * x[:, 0] * x[:, 1] + x[:, 1] ** 2
+
+
+def f_target_hard(x):
+    return 1.0 + 1.0 / (x[:, 0] + x[:, 1])
+
+
+# %%
+# Then we define the objective function for the evolution. It uses the
+# mean-squared error between the expression represented by a given
+# individual and the target function evaluated on a set of random
+# points.
+
+
+def objective(individual, target_function, seed):
+    """Objective function of the regression task.
+
+    Parameters
+    ----------
+    individual : Individual
+        Individual of the Cartesian Genetic Programming Framework.
+    target_function : Callable
+        Target function.
+
+    Returns
+    -------
+    Individual
+        Modified individual with updated fitness value.
+    """
+    if not individual.fitness_is_None():
+        return individual
+
+    n_function_evaluations = 1000
+
+    np.random.seed(seed)
+
+    f_graph = individual.to_numpy()  # generate a NumPy compatible function
+    x = np.random.uniform(-4, 4, size=(n_function_evaluations, 2))
+    with warnings.catch_warnings():  # ignore warnings due to zero division
+        warnings.filterwarnings("ignore", message="divide by zero encountered in true_divide")
+        warnings.filterwarnings("ignore", message="invalid value encountered in")
+        try:
+            y = f_graph(x[:, 0], x[:, 1])  # pass a vector of values to the function
+        except ZeroDivisionError:
+            individual.fitness = -np.inf
+            return individual
+
+    loss = np.mean((target_function(x) - y) ** 2)
+    individual.fitness = -loss
+
+    return individual
+
+
+# %%
+# Next, we define the main loop of the evolution. To easily execute it
+# for different target functions, we wrap it into a function here. It
+# comprises:
+#
+# - defining the parameters for the population, the genome of individuals,
+#   and the evolutionary algorithm.
+# - creating a Population instance and instantiating the evolutionary algorithm.
+# - defining a recording callback closure for bookkeeping of the progression of the evolution.
+#
+# Finally, we call the `evolve` method to perform the evolutionary search.
+
+
+def evolution(f_target):
+    """Execute CGP on a regression task for a given target function.
+
+    Parameters
+    ----------
+    f_target : Callable
+        Target function
+
+    Returns
+    -------
+    dict
+        Dictionary containing the history of the evolution
+    Individual
+        Individual with the highest fitness in the last generation
+    """
+    population_params = {"n_parents": 10, "seed": 818821}
```

Comment:
Is there a reason to not use the default parameterization? (Except for `genome_params['n_inputs] = 2` ?

Reply:
no reason except staying close to the "evo_regression" example. i'd like to keep it this way to really make minimal changes to the existing one, but i created a new issue that we should update the examples to use default parameters where possible.